js/main.js

```diff
@@ -13,155 +10,194 @@ var titles = [
   'Квартира в центре, элитный ЖК',
   'Есть ещё пара свободных квартир, по всем вопросам с удовольствием отвечу'
 ];
-var photosList = [
+var TYPE_HOUSINGS = ['palace', 'flat', 'house', 'bungalo'];
+var REGISTRATION_TIMES = ['12:00', '13:00', '14:00'];
+var FEATURES = ['wifi', 'dishwasher', 'parking', 'washer', 'elevator', 'conditioner'];
+var PHOTOS = [
   'http://o0.github.io/assets/images/tokyo/hotel1.jpg',
   'http://o0.github.io/assets/images/tokyo/hotel2.jpg',
   'http://o0.github.io/assets/images/tokyo/hotel3.jpg'
 ];
+var TYPE_HOUSING = {
+  flat: 'Квартира',
+  bungalo: 'Бунгало',
+  house: 'Дом',
+  palace: 'Дворец'
+};
+var GUEST_ROOM = {
+  '1': {
+    'guests': ['1'],
+    'errorText': '1 комната для 1 гостя'
+  },
+  '2': {
+    'guests': ['1', '2'],
+    'errorText': '2 комнаты для 1 или 2 гостей'
+  },
+  '3': {
+    'guests': ['1', '2', '3'],
+    'errorText': '3 комнаты для 1, 2 или 3 гостей'
+  },
+  '100': {
+    'guests': ['0'],
+    'errorText': '100 комнат не для гостей'
+  }
+};
+var BEGINNING_MAP_X = 0;
+var END_MAP_X = 1200;
+var BEGINNING_MAP_Y = 130;
+var END_MAP_Y = 630;
+var PIN_WIDTH = 50;
+var PIN_HEIGHT = 70;
+var NUMBER_ADS = 8;
+var map = document.querySelector('.map');
+var mapPinsElement = map.querySelector('.map__pins');
+var mapFiltersContainer = map.querySelector('.map__filters-container');
+var mapPinMain = mapPinsElement.querySelector('.map__pin--main');
+var adForm = document.querySelector('.ad-form');
+var adFormFieldsets = adForm.querySelectorAll('fieldset');
+var mapFiltersElement = map.querySelector('.map__filters');
+var addressInput = adForm.querySelector('#address');
+var roomNumberSelect = adForm.querySelector('#room_number');
+var capacitySelect = adForm.querySelector('#capacity');
 
 //  Добавляет 0 перед числом
 var addZero = function (num) {
-  if (num < 10) {
-    num = '0' + num;
-  }
-
-  return num;
+  return num < 10 ? '0' + num : num;
 };
 
 //  Находит случайное число в интервале
 var getRandomInt = function (min, max) {
   return Math.floor(Math.random() * (max - min + 1)) + min;
 };
 
+//  Возвращает случайный элемент из массива
+var getRandomElement = function (elements) {
+  return elements[getRandomInt(0, elements.length - 1)];
+};
+
 //  Вырезает нужное количество элементов из массива
-var cutElements = function (arr, length) {
-  return arr.splice(0, length);
+var cutElements = function (elements) {
+  var length = getRandomInt(1, elements.length);
+  return elements.slice(0, length);
 };
 
 //  Перемешивает массив
-var shufflesArray = function (arr) {
-  var result = [];
-  var randomIndex;
-  var elem;
-  var arr1 = arr.slice(0);
-  while (arr1.length > 0) {
-    randomIndex = getRandomInt(0, arr1.length - 1);
-    elem = arr1.splice(randomIndex, 1)[0];
-    result.push(elem);
+var shuffleArray = function (elements) {
+  var copyElements = elements.slice();
+  for (var i = copyElements.length - 1; i > 0; i--) {
+    var j = Math.floor(Math.random() * (i + 1));
```

Comment:
Нужно брать случайный индекс из всего диапазона, можно использовать функцию getRandomInt
```suggestion
    var j = getRandomInt(0, copyElements.length - 1);
```
Лучше сохрани в переменную copyElements.length перед циклом и используй переменную